apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -124,7 +124,7 @@ def __init__(
         # build indexers, as needed
         self.axes_to_reindex = set(range(self.ndim)) - set(self.reindex_disable_on_axis)
         self.minor_axes_indexer = {
-            d: pd.Index(self.joinids[d].to_numpy())
+            d: build_index(self.joinids[d].to_numpy())
```

Comment:
I suspect this should pull the concurrency from the current context, not just take the default (as is done in ExperimentAxisQuery).  Unfortunately, the context is not passed down to the iterator (it is accessible a couple of calls up the stack).